### ISSUE TITLE
flake8 2.x plugins break trying to accept a `builtins` argument in 3.x.

### ISSUE THREAD

**Issue by asottile** [PyCQA/flake8#506]
In GitLab by @tomprince on Jun 27, 2016, 15:06

There are a bunch of existing plugins (at least`flake8-mock`, `flake8-debugger`, `flake8-commas`), that take a `builtins` argument in their `__init__`. This causes checking to hand with the following output:
```
Process Process-1:
Traceback (most recent call last):
  File "/usr/lib64/python2.7/multiprocessing/process.py", line 258, in _bootstrap
    self.run()
  File "/usr/lib64/python2.7/multiprocessing/process.py", line 114, in run
    self._target(*self._args, **self._kwargs)
  File "/home/tomprince/.virtualenvs/buildbot/lib/python2.7/site-packages/flake8/checker.py", line 222, in _run_checks_from_queue
    checker.run_checks(self.results_queue, self.statistics_queue)
  File "/home/tomprince/.virtualenvs/buildbot/lib/python2.7/site-packages/flake8/checker.py", line 548, in run_checks
    self.run_ast_checks()
  File "/home/tomprince/.virtualenvs/buildbot/lib/python2.7/site-packages/flake8/checker.py", line 455, in run_ast_checks
    checker = self.run_check(plugin, tree=ast)
  File "/home/tomprince/.virtualenvs/buildbot/lib/python2.7/site-packages/flake8/checker.py", line 434, in run_check
    self.processor.keyword_arguments_for(plugin.parameters, arguments)
  File "/home/tomprince/.virtualenvs/buildbot/lib/python2.7/site-packages/flake8/processor.py", line 213, in keyword_arguments_for
    arguments[param] = getattr(self, param)
AttributeError: 'FileProcessor' object has no attribute 'builtins'
```

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Jun 27, 2016, 16:30

That's very surprising. We've never supported that as a parameter. None of those plugins appear to be actually using it either. It also appears they all copied and pasted a bunch of code from somewhere for I don't know what reason. =/

I can fix this to not raise an AttributeError, but those plugins should also be fixed. Thanks for reporting this @tomprince

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Jun 28, 2016, 07:39

Status changed to closed by commit 2d3e277b1e0c0a24c30c611558692f95d14a8470

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Jun 30, 2016, 12:57

mentioned in commit ahal/flake8@2d3e277b1e0c0a24c30c611558692f95d14a8470
